### DISCUSSION TITLE
Java support?

### DISCUSSION THREAD

**Discussion by samDobsonDev** [BoundaryML/baml#885] [Ideas]
Any consideration for Java support any time soon? In the meantime, or if that's off the table, is there any way I can still utilise BAML in a Java application without specific API support, or any other language for that matter?

**Comment by aaronvg** [comment]
we have an idea of supporting a baml server you can setup that you could deploy via docker/etc, and we can generate a Java client using an OpenAPI spec (also based on your function definitions) to call that server/your functions. Does that approach sound workable to you?

**Comment by samDobsonDev** [comment]
Yeah for sure, that would be perfect!

**Reply by aaronvg** [reply]
sounds good, we'll get back to you on this.

**Comment by hellovai** [comment]
@samDobsonDev this is ready now!

Check out https://docs.boundaryml.com/docs/get-started/quickstart/openapi

Let me know if you ahve any questions!
